### PR TITLE
AWS bucket listing

### PR DESCRIPTION
**Pull request by rajnikantsh**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3030)

### CODE REVIEW

File: aws/src/main/python/s3BucketList/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                            } ])
+    return res['Instances'][0]['InstanceId']
+
+def lambda_handler(event, context):
+    region = event.get('region', DEFAULT_REGION)
+    size = event.get('size', 100)
+    shutdown_behaviour = 'terminate'
+    shutdown_wait = "10"
+    init_ec2(region)
+    script = initscript.replace("$N", size).replace('$REGION', region).replace('$SHUTDOWN_WAIT', shutdown_wait)
```
Comment (JustinPihony) [inline aws/src/main/python/s3BucketList/lambda_function.py:70]: You have to turn the size into a string explicitly and use single quotes I found:

```
script = initscript.replace('$N', str(size)).replace('$REGION', region).replace('$SHUTDOWN_WAIT', shutdown_wait)
```

**Review by JustinPihony — COMMENTED** [review]
A couple of fixes needed before it can run (already applied to the lambda since that is how I was testing). That said, any idea of how long this should be expected to take? I had it running for an hour with no output.

**Comment by JustinPihony** [comment]
My attempt to run this ended in:

> [Errno 32] Broken pipe
> Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='UTF-8'>

So, I'm wondering if this needs to be optimized and run in a different way?

File: aws/src/main/python/s3BucketList/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import datetime
+
+initscript = (('''#cloud-config
+runcmd:
+  - echo "-------------------checking bucket size----------------------"
+  - sudo dpkg --configure -a
+  - sudo dpkg --remove --force-remove-reinstreq unattended-upgrades
+  - sudo apt-get install unattended-upgrades
+  - sudo dpkg --configure -a
+  - sudo apt update
```
[the remark this answers is not in the document]
Comment (rajnikantsh) [reply aws/src/main/python/s3BucketList/lambda_function.py:14]: applied this change

File: aws/src/main/python/s3BucketList/lambda_function.py (hunk 1 above)
Comment (rajnikantsh) [reply aws/src/main/python/s3BucketList/lambda_function.py:77]: applied this change

**Comment by rajnikantsh** [comment]
> A couple of fixes needed before it can run (already applied to the lambda since that is how I was testing). That said, any idea of how long this should be expected to take? I had it running for an hour with no output.

@JustinPihony It will take around 5 hours to complete. Meanwhile you can check internediatory results in `/result.csv` . That other issue related to broken pipe has been fixed

**Review by JustinPihony — APPROVED** [review]
Looks good, but has it been tested? Also, has https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-inventory.html been looked into?

**Comment by rajnikantsh** [comment]
> Looks good, but has it been tested? Also, has https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-inventory.html been looked into?

yes its tested one time and results is on location : https://beam-outputs.s3.us-east-2.amazonaws.com/s3-buckets-output.csv 

also I have reviewed https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-inventory.html . It can be used in our case if we can identify that these bucket/folder we have to monitor. In current solution we dont need to know bucket/folder name. It will list top N folder by size

**Comment by JustinPihony** [comment]
test!

**Pull request merged**